nRF5SDK_mods/clang/nrf_svci.h

```diff
@@ -0,0 +1,305 @@
+        static __INLINE return_type function_name(p0t p0n, p1t p1n, p2t p2n, p3t p3n)               \
+        {                                                                                           \
+            return svci_ ## function_name(svci_num, p0n, p1n, p2n, p3n);                            \
+        }
+        
+#else
+        
+#if defined (__GNUC__)
+
+    #define SVCI_DECL_0(svci_num, return_type, function_name)           \
+        _Pragma("GCC diagnostic push")                                  \
+        _Pragma("GCC diagnostic ignored \"-Wreturn-type\"")             \
+        __attribute__((naked, unused))                                  \
+        static return_type svci_ ## function_name(void)                 \
+        {                                                               \
+            /* Do the SuperVisor call by using svc instruction with     \
+               R12 containing the SVCI number */                        \
+            __ASM __volatile                                            \
+            (                                                           \
+                "  ldr r12, =%0       \n"                               \
+                "  svc %1             \n"                               \
+                "  bx lr              \n"                               \
+                "  .ltorg"                                              \
+                : /* output */                                          \
+                : /* input */                                           \
+                    "X"(svci_num),                                      \
+                    "I"(GCC_CAST_CPP NRF_SVCI_SVC_NUM)                  \
+                : /* clobbers */                                        \
+                    "r12"                                               \
+            );                                                          \
+        }                                                               \
+        _Pragma("GCC diagnostic pop")    
+    
+    #define SVCI_DECL(svci_num, return_type, function_name, ...)        \
+        _Pragma("GCC diagnostic push")                                  \
+        _Pragma("GCC diagnostic ignored \"-Wreturn-type\"")             \
+        __attribute__((naked, unused))                                  \
+        static return_type svci_ ## function_name(__VA_ARGS__)          \
+        {                                                               \
+            /* Do the SuperVisor call by using svc instruction with     \
+               R12 containing the SVCI number */                        \
+            __ASM __volatile                                            \
+            (                                                           \
+                "  movw r12, %0       \n"                               \
+                "  movt r12, %1       \n"                               \

```

Comment:
If this is the only change from `ldr.w`, it's possible this is now supported in LLVM.

Looking at this patch from 2021 and the test files included it should work now?
Could you try reverting this file to the original and see if that works? If so, we can remove this file from the PR.

Reply:
Running with Clang-14 (Current docker build version) we still get this error on removing the file. Clang-14 came out in 2022 so I'd imagine it should have this patch. Maybe something else is going on? Seems to be complaining about the '#' but not ldr.w.

![image](https://github.com/microbit-foundation/codal-microbit-nrf5sdk/assets/47667973/c71fe41e-5d75-410c-9359-cab006e5c6ce)

Reply:
Same error... Do you think this could be of interest? https://github.com/lancaster-university/codal-nrf52/blob/master/asm/CortexContextSwitch.s#L26

Reply:
Ah! I think `.thumb` is right, but that is setting the cpu mode to cortex m0, which is armv6m and doesn't have full thumb2 support. If the `.cpu cortex-m0` line is changed to `.cpu cortex-m4` does that work?

Reply:
No change, would be great if we could find what it causing it to be upset. In other news I was trying the alternative patch we discussed and ended up accidentally applying none of the changes aside from app_error_handler_gcc.c and nrf_nvci.h. This somehow ran to completion on a complete LLVM build. We may not need all of these changes! Not sure what has changed from a few months ago however. Still need to test linking against GCC libraries. But if that works...

Reply:
Some of the BLE might only be compiled if the additional codal flags are added:
https://github.com/lancaster-university/microbit-v2-samples/blob/master/codal.ble.json

By also running this function in main() we can ensure all the BLE services are included:
https://github.com/lancaster-university/microbit-v2-samples/blob/master/source/samples/BLETest.cpp#L50

If that compiles and runs, then that's be great news!

Reply:
[MICROBIT.zip](https://github.com/microbit-foundation/codal-microbit-nrf5sdk/files/12622348/MICROBIT.zip)

Followed the steps for running ble_test(); and with partial flash disabled I get the above output. Not sure on what to expect so was wondering if you could run it through some tests but it shows up on other devices and I can connect to it.

Reply:
Looks great, the programme runs and I tested it successfully with https://thegecko.github.io/microbit-web-components/examples/index.html

🎉